enhancements/security/x509-trust.md

```diff
@@ -0,0 +1,213 @@
+
+The cluster-version operator (CVO) [merges ConfigMaps][cluster-version-operator-EnsureConfigMap] by [clobbering any manifest-defined data][cluster-version-operator-mergeMap], [labels, and annotations][cluster-version-operator-EnsureObjectMeta].
+Data keys, labels, and annotations not defined in the manifest are ignored.
+This means that components which decide to provide their per-component ConfigMaps via the CVO should not set labels like `config.openshift.io/inject-default-cabundle` unless they also set [the `release.openshift.io/create-only` annotation to `true`][cluster-version-operator-create-only], to allow cluster administrators to manipulate the per-component ConfigMap without the CVO stomping on their updates.
+For example, the registry operator [sets `create-only`][registry-configmap-create-only].
+
+#### Universal Base Image Containers
+
+FIXME: UBI-specific docs for this [RHEL behavior](rhel-cert-injection) that show that you don't need to install some non-UBI RHEL stuff to make it work.
+
+Components can mount `ca-bundle.crt` from their ConfigMap(s) under `/usr/share/pki/ca-trust-source/anchors/`.
+Components are then responsible for running `update-ca-trust` (FIXME: do we need `extract` or not?) at start.
+They must also run `update-ca-trust` periodically thereafter to pull in any [updates from the mounted ConfigMap][mounted-configmap-updates].
+
+Components that need a single trust bundle injected can bypass `update-ca-trust` and mount their `ca-bundle.crt` directly into `/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem` (FIXME: RHEL/UBI docs confirming this as a supported workflow).
+For example, the registry operator [does this][registry-configmap-mount].
+
+FIXME: We need to grow a way to handle `override` `composition`.
+
+### Node Trust
+
+[Red Hat Enterprise Linux CoreOS (RHCOS)][rhcos], and nodes more broadly, have a separate trust-management framework getting hashed out [here][machine-config-operator-trust].
+
+### Container Image Registry Trust
+
+FIXME: fill in how this works
+
+### User Stories
+
+#### Transparent proxies
+
+FIXME
+
+### Implementation Details/Notes/Constraints
+
+The network operator supports copying trust bundles between ConfigMaps based on labels on the target ConfigMaps.
+
+* [The `config.openshift.io/inject-default-cabundle` label](../proxy/global-cluster-egress-proxy.md#implementation-detailsnotesconstraints-optional) asks for the current [cluster-scoped default trust bundle](#cluster-scoped-default-trust).
+* [The `config.openshift.io/inject-proxy-cabundle` label](../proxy/global-cluster-egress-proxy.md#implementation-detailsnotesconstraints-optional) asks for the current proxy trust bundle.
+    The network operator will fall back to [the cluster-scoped default trust bundle](#cluster-scoped-default-trust) if no proxy-specific trust bundle has been configured.
```

Comment:
Do you plan to grow this list of labels in the future or even build it dynamically if it's possible at all? This could be useful if we want to keep cert bundles separate for different usages.

Reply:
> Do you plan to grow this list of labels in the future...

Yes.

> ... or even build it dynamically if it's possible at all?

No.  The network operator can experiment.  Once we want to commit to support, we can freeze labels out here and in openshift-docs.